### PR TITLE
Fix #1760, check resource ID idx is less than max

### LINKED ISSUE #1760: Write Resource ID functional test to check maximum
**Is your feature request related to a problem? Please describe.**
Need open source functional tests for certifiability

**Describe the solution you'd like**
Add additional checks to functional tests for ES Resource ID to Index that ensure the returned index is less than the maximum allowed for that resource type

**Describe alternatives you've considered**
None

**Requester Info**
_Originally posted by @astrogeco in https://github.com/nasa/cFE/pull/1734#issuecomment-892799313_

### PR DESCRIPTION
**Pull request by nmullane**
**Describe the contribution**
Fixes #1760 

**Testing performed**
Steps taken to test the contribution:
1. Build and run cfe functional tests

**Expected behavior changes**
No impact to behavior

**System(s) tested on**
 - Hardware: PC
 - OS: Ubuntu 20.04

**Contributor Info - All information REQUIRED for consideration of pull request**
Niall Mullane - GSFC 582 Intern

### CODE REVIEW

**Comment by astrogeco** [comment]
[CCB:2021-08-11](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-August-11-2021) APPROVED with CHANGES
- Add call to "unregister"

**Comment by astrogeco** [comment]
[CCB:2021-08-11](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-August-11-2021) APPROVED

File: modules/cfe_testcase/src/es_resource_id_test.c [hunk 1]
```diff
@@ -89,6 +92,7 @@ void TestCounterIDToIndex(void)
     UtAssert_UINT32_EQ(CFE_ES_RegisterGenCounter(&CounterId, CounterName), CFE_SUCCESS);
```
Comment (skliper) [inline modules/cfe_testcase/src/es_resource_id_test.c:92]: Unregister counter at end of test

**Review by skliper — CHANGES_REQUESTED** [review]
Just adding the request to change so it doesn't get lost.

File: modules/cfe_testcase/src/es_resource_id_test.c (hunk 1 above)
Comment (nmullane) [reply modules/cfe_testcase/src/es_resource_id_test.c:92]: done 👍

**Review by skliper — APPROVED** [review]
